### PR TITLE
Add Interface for Game Invites

### PR DESCRIPTION
**Pull request by catornot**
apparently game invites still work but only the invites so the lobbies infrastructure isn't maintained by discord.

So discord rpc now exposes a interface for it :)

I have made a plugin that uses the interface already : https://github.com/catornot/compartya 

[how to test](https://discord.com/channels/920776187884732556/950322078945538058/1275578960566026301)

### CODE REVIEW

**Comment by GeckoEidechse** [comment]
I haven't been able to try out that PR yet, how do the invites actually work?

For the Discord side, I guess it uses https://support.discord.com/hc/en-us/articles/115001557452-Game-Invites and from the Northstar side it just tells the client to just join a specific server id?

File: Cargo.lock
Comment (GeckoEidechse) [inline Cargo.lock:1]: Note to self to add dependabot to this repo so that we don't need to deal with manual version bumps ^^

Comment (GeckoEidechse) [reply Cargo.lock:1]: - #23

**Comment by catornot** [comment]
> For the Discord side, I guess it uses https://support.discord.com/hc/en-us/articles/115001557452-Game-Invites and from the Northstar side it just tells the client to just join a specific server id?

The pr only handles discord invites and everything else is delegated to another plugin which is why there is an interface for it. It doesn't have to be the compartya plugin btw.

File: Cargo.lock
Comment (GeckoEidechse) [reply Cargo.lock:1]: Unrelated to this PR.

**Comment by catornot** [comment]
should work
rustc approved :+1:

**Comment by GeckoEidechse** [comment]
I spliced out the `rrplug` `4.0.0` related changes into #24 so that we can get them merged and released faster as part of the next patch release.

**Comment by catornot** [comment]
ok, I will be waiting

**Comment by GeckoEidechse** [comment]
> ok, I will be waiting

no waiting, testing :point_right::point_left:

**Comment by catornot** [comment]
> > ok, I will be waiting
> 
> no waiting, testing 👉👈

well I still need to wait to test it XD

**Comment by Zanieon** [comment]
How do I test this? I replaced my dll with this PR but Discord is not showing invites, also is there any squirrel hooks for this? Because if this gets in, i might update my Matchmaking Mixtape Menu to make use of this feature and restore the "Invite Friends" button in the main lobby.

**Comment by catornot** [comment]
> How do I test this? I replaced my dll with this PR but Discord is not showing invites, also is there any squirrel hooks for this? Because if this gets in, i might update my Matchmaking Mixtape Menu to make use of this feature and restore the "Invite Friends" button in the main lobby.

The pr just adds interfaces to this plugin so other plugins could handle invites.

In the description I did mention a plugin that uses the interface :>

**Comment by catornot** [comment]
can this pr just be merged?
image

**Comment by EM4Volts** [comment]
Tested, works. thanks.

**Review by GeckoEidechse — APPROVED** [review]
Confirmed working in testing.

First of all game didn't crash and DiscordRPC still worked.

Tested compartya together with @catornot with this PR.

After some initial bugs with compartya (couldn't resolve local IP due to simplistic `ipconfig` parsing) we got it working and @catornot joined my party and got pulled into servers I joined and similarly I got pulled into servers when I joined @catornot's party.

**Comment by GeckoEidechse** [comment]
(still not a fan of updating `Cargo.lock` in the same PR but oh well)

**Pull request merged**